core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java

```diff
@@ -893,4 +893,33 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.endObject();
         return builder;
     }
+
+    @Override
+    public String toString() {
+        StringBuilder result = new StringBuilder();
+        String newLine = System.getProperty("line.separator");
+
+        result.append(this.getClass().getName());
+        result.append(" Object {");
+        result.append(newLine);
+
+        Field[] fields = this.getClass().getDeclaredFields();
```

Comment:
There's no need for reflection here - writing out all the fields, in a sensible order, is much preferred.